services/code_hosting/git_hub_service.rb

```diff
@@ -36,6 +36,36 @@ def username
       client.login
     end
 
+    # returns all open pull requests on given repo
+    def open_pull_requests(repo_full_name: nil)
+      return client.pull_requests(repo_full_name, state: "open")
+    end
+
+    # returns only the most recent commit_sha for every open pr
+    def last_commit_sha_for_all_open_pull_requests(repo_full_name: nil)
+      return self.open_pull_requests(repo_full_name: repo_full_name).map { |pull_request| pull_request.head.sha }
+    end
+
+    # returns the status of a given commit sha for a given repo
+    def status_for_commit_sha(repo_full_name: nil, sha: nil)
+      return client.statuses(repo_full_name, sha)
+    end
+
+    # updates the most current commit to "pending" on all open prs if they don't have a status.    
+    # returns a list of commits that have been updated to `pending` status
+    def update_all_open_prs_without_status_to_pending_status!(repo_full_name: nil)
+      open_pr_commits = self.last_commit_sha_for_all_open_pull_requests(repo_full_name: repo_full_name)
+      updated_commits = []
+      open_pr_commits.each do |sha|
+        statuses = self.status_for_commit_sha(repo_full_name: repo_full_name, sha: sha)
```

Comment:
Does this fetch all statuses, or just the statuses created by `fastlane.ci`?

Reply:
alllll

Reply:
Let me append a question here: should we automatically remove all non-fastlane-ci build status in the `status_for_commit_sha` method, to make things work?

Reply:
I have a TODO: in some code I'm working on right now.
Yes, we should filter out statuses not created by fastlane.ci 🚫

Reply:
perfect, thanks 👍